server/src/main/java/org/elasticsearch/index/mapper/Mapper.java

```diff
@@ -132,10 +136,14 @@ public ParserContext createMultiFieldContext(ParserContext in) {
                 return new MultiFieldParserContext(in);
             }
 
+            public String getDataStreamTimestampField() {
+                return dataStreamTimestampField;
+            }
+
             static class MultiFieldParserContext extends ParserContext {
                 MultiFieldParserContext(ParserContext in) {
                     super(in.similarityLookupService(), in.mapperService(), in.typeParsers(),
-                            in.indexVersionCreated(), in.queryShardContextSupplier());
+                            in.indexVersionCreated(), in.queryShardContextSupplier(), null);
```

Comment:
I think we should pass through the timestamp field here. I guess the timestamp could happen to be a multi-field.

Reply:
I will pass down the timestamp field, a small note, the timestamp field can only be a field that is part of the _source. 
There is validation that checks whether a field mapping exists of type date or date_nanos when creating the composable index template and this is also asserted when a backing index of a data stream is created.

Reply:
> the timestamp field can only be a field that is part of the _source.

Interesting! I am generally curious to catch up on timestamp mapping validation, I'll ping the team offline about this.